### PR TITLE
Saithat/h4 ish 8 parse query

### PR DESCRIPTION
**Pull request by saithat**
We'll probably need more test cases since this regex might not be robust enough.

### CODE REVIEW

File: src/helpers/ParseQuery.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * @Param str The string to parse
+ * @Returns A string telling what the input is
+ */
+export default function Parse(str: string): string {
+  const zip = new RegExp('^\\d{5}$');
+  // let county = new RegExp('(.+), ([a-z]{2})');
+  // let county: RegExp = /(.+), ([a-z]{2})/;
+  const county = new RegExp('(^[A-z]+), ([A-z]{2})');
+  const add1 = new RegExp('(.+), ([A-z]{2})');
+  // let add2: RegExp = /\d{1,6}\s(?:[A-Za-z0-9#]+\s){0,7}(?:[A-Za-z0-9#]+,)\s*(?:[A-Za-z]+\s){0,3}(?:[A-Za-z]+,)\s*[A-Z]{2}\s*\d{5}/;
```
Comment (VishalAiely) [inline src/helpers/ParseQuery.ts:12]: I think having the simple regex for now that can pick between the three for most cases will be good. We can expand on it later if needed

**Review by VishalAiely — DISMISSED** [review]
Everything seems good to me and the test cases cover good examples 👍

**Review by VishalAiely — APPROVED** [review]

File: src/helpers/ParseQuery.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  console.log('zipcode: ', zip.test(str));
+  console.log('county: ', county.test(str));
+  console.log('add1: ', add1.test(str));
```
Comment (huncfut) [inline src/helpers/ParseQuery.ts]: Should we keep the console logs in the master branch?
```suggestion
```

File: src/helpers/ParseQuery.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    return 'county';
+  }
+  if (add1.test(str)) {
+    return 'add1';
+  }
+  // if (add2.test(str)){
+  //    return 'add2';
+  // }
+
+  return 'error';
```
Comment (huncfut) [inline src/helpers/ParseQuery.ts]: It definitely works very well. I just have a suggestion that will make it a bit more readable (at least to me), scalable, and condense.
```suggestion
  return zip.test(str) && 'zip'
    || county.test(str) && 'county'
    || add1.test(str) && 'add1'
    || 'error'
```

File: src/helpers/ParseQuery.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ * Returns what the input is (zip code, county name, or address)
+ * @Param str The string to parse
+ * @Returns A string telling what the input is
+ */
+export default function Parse(str: string): string {
+  const zip = new RegExp('^\\d{5}$');
+  // let county = new RegExp('(.+), ([a-z]{2})');
+  // let county: RegExp = /(.+), ([a-z]{2})/;
+  const county = new RegExp('(^[A-z]+), ([A-z]{2})');
+  const add1 = new RegExp('(.+), ([A-z]{2})');
```
Comment (huncfut) [inline src/helpers/ParseQuery.ts:11]: It took me a while before I got what add1 stands for (later in the code). I think that it would be easier for us in the future to make these tags self-explanatory (if that is a word). Maybe 'address' for now?
```suggestion
  const address = new RegExp('(.+), ([A-z]{2})');
```

**Review by huncfut — COMMENTED** [review]
Everything seems to be good. I just made a few suggestions – you decide what to do!

File: src/helpers/ParseQuery.ts (hunk 2 above)
Comment (saithat) [reply src/helpers/ParseQuery.ts]: Oh, I thought I thought I had removed the console.logs!

File: src/helpers/ParseQuery.ts (hunk 4 above)
Comment (saithat) [reply src/helpers/ParseQuery.ts:11]: Sure! I left it add1 because I was thinking we might add a couple of more regex for more coverage later on at some point.

File: src/helpers/ParseQuery.ts (hunk 3 above)
Comment (saithat) [reply src/helpers/ParseQuery.ts]: Hmm it's definitely more concise and elegant, but I think how it's written before is a bit more readable since it's simpler.  I'm not sure it is more scalable though. Doesn't each part in a boolean expression still have to get executed in order to take advantage of the short-circuiting?

Comment (huncfut) [reply src/helpers/ParseQuery.ts]: I am not a sure, but I think that it breaks as soon as one && statement is reached. It's probably only the preference at this point. I don't believe there is one right way to do it. I think you should do whatever you think will work best for our team.

Comment (Riz-waan) [reply src/helpers/ParseQuery.ts]: The before version seems much more clearer on what the conditions are but I feel like @huncfut version seems to make it clear that you are returning only one value and that value is dependent on the different conditions.

Comment (Riz-waan) [reply src/helpers/ParseQuery.ts]: I also think it should be consistent throughout the team. It would be hard to follow if one person used if statements and someone else used the concise version.

**Review by Riz-waan — COMMENTED** [review]
```
// let county = new RegExp('(.+), ([a-z]{2})');
// let county: RegExp = /(.+), ([a-z]{2})/;
```

What are these lines and why are they commented?

Also, I agree with @huncfut that add1 should be more descriptive. I also understand what you mean too. Maybe something like standardAddress or address1?

**Comment by saithat** [comment]
> ```
> // let county = new RegExp('(.+), ([a-z]{2})');
> // let county: RegExp = /(.+), ([a-z]{2})/;
> ```
> 
> What are these lines and why are they commented?
> 
> Also, I agree with @huncfut that add1 should be more descriptive. I also understand what you mean too. Maybe something like standardAddress or address1?

Those lines were just regex I tested before for checking county. I left them because I thought we might have to come back to this code once we have more realistic test cases. But, I can delete both. Yeah, I'll change it to address1!

File: src/helpers/ParseQuery.ts (hunk 3 above)
Comment (VishalAiely) [reply src/helpers/ParseQuery.ts]: I think having the if statements is a bit more readable in this scenario but the other way would work the same

File: src/helpers/ParseQuery.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+/**
+ * Returns what the input is (zip code, county name, or address)
+ * @Param str The string to parse
+ * @Returns A string telling what the input is
+ */
+export default function Parse(str: string): string {
```
Comment (VishalAiely) [inline src/helpers/ParseQuery.ts:6]: One thing I forgot to mention earlier was that you could change the return type here from `string` to `'zip' | 'country' | 'address' | 'error'` to make it a bit more explicit. If it looks like a lot to put on that line you could make a new type called `SearchQueryType` or something similar and use it here

**Review by huncfut — APPROVED** [review]

**Review by braxtonhaynie — APPROVED** [review]
